snippets/base/models.py

```diff
@@ -258,7 +262,35 @@ def __unicode__(self):
         return self.description
 
 
-class Snippet(CachingMixin, models.Model):
+class SnippetBaseModel(models.Model):
+    def duplicate(self):
+        snippet_copy = copy.copy(self)
+        snippet_copy.id = None
+        snippet_copy.disabled = True
+        snippet_copy.name = '{0} - {1}'.format(
+            self.name,
+            datetime.strftime(datetime.now(), '%Y.%m.%d %H:%M:%S'))
+        snippet_copy.save()
+
+        for field in self._meta.get_all_field_names():
```

Comment:
I'm not sure I'm a fan of the generic approach here. I think it'd be better to have the subclasses override this to handle copying their related things directly, like the locales. Thoughts?